Sources/NimiqClient/NimiqClient.swift

```diff
@@ -0,0 +1,1082 @@
+    /// - Returns: Array of Accounts owned by the client.
+    public func accounts() throws -> [Any]? {
+        let result: [RawAccount] = try call(method: "accounts", params: [])!
+        var converted: [Any] = [Any]()
+        for rawAccount in result {
+            converted.append(rawAccount.value)
+        }
+        return converted
+    }
+
+    /// Returns the height of most recent block.
+    /// - Returns: The current block height the client is on.
+    public func blockNumber() throws -> Int? {
+        return try call(method: "blockNumber", params: [])
+    }
+
+    /// Returns information on the current consensus state.
+    /// - Returns: Consensus state. `established` is the value for a good state, other values indicate bad.
+    public func consensus() throws -> ConsensusState? {
+        return try call(method: "consensus", params: [])
+    }
+
+    /// Returns or overrides a constant value.
+    /// When no parameter is given, it returns the value of the constant. When giving a value as parameter,
+    /// it sets the constant to the given value. To reset the constant use `resetConstant()` instead.
+    /// - Parameter string: The class and name of the constant (format should be `Class.CONSTANT`).
+    /// - Parameter value: The new value of the constant.
+    /// - Returns: The value of the constant.
+    public func constant(_ constant: String, value: Int? = nil) throws -> Int? {
+        var params:[Any] = [constant]
+        if value != nil {
+            params.append(value!)
+        }
+        return try call(method: "constant", params: params)
+    }
+
+    /// Creates a new account and stores its private key in the client store.
+    /// - Returns: Information on the wallet that was created using the command.
+    public func createAccount() throws -> Wallet? {
+        return try call(method: "createAccount", params: [])
+    }
+
+    /// Creates and signs a transaction without sending it. The transaction can then be send via `sendRawTransaction()` without accidentally replaying it.
+    /// - Parameter transaction: The transaction object.
+    /// - Returns: Hex-encoded transaction.
+    public func createRawTransaction(_ transaction: OutgoingTransaction) throws -> String? {
+        let params:[String:Any?] = [
+            "from": transaction.from,
+            "fromType": transaction.fromType?.rawValue,
+            "to": transaction.to,
+            "toType": transaction.toType?.rawValue,
+            "value": transaction.value,
+            "fee": transaction.fee,
+            "data": transaction.data
+        ]
+        return try call(method: "createRawTransaction", params: [params])
+    }
+
+    /// Returns details for the account of given address.
+    /// - Parameter address: Address to get account details.
+    /// - Returns: Details about the account. Returns the default empty basic account for non-existing accounts.
+    public func getAccount(address: Address) throws -> Any? {
```

Comment:
nitpick (non-blocking): this could be refactored to have "fluent design". 

This applies to a lot of methods in this class. This one could be refactored to `public func account(for address: Address)` which would look more like a Swift API design. It would result in a call like this `account(for: "NQ...")`.
Source: [Strive for Fluent Usage](https://swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage).

Reply:
Thanks for pointing that out. Back in the days of ObjC, and now with Swift, I've always liked the self documented style in the APIs. However, with the Nimiq RPC clients I tried not to change the methods names to fit the language naming conventions, because people familiar with the original API specs wouldn't know what method to use for each call. I'm going to check other JSON RPC implementations in Swift to see what approach has been taken and will start to refactor the code in case that the Swift conventions are preferred, also will add a note in each method's doc with the exact name as it's in the original API specs for clarification.

Reply:
The code has been refactored, it now adheres to the Fluent Usage naming recommendations, however the `get` prefix was kept because it seems a convention in the original specification to indicate whether the information is available as a state in the node or rather it has to be fetched from the blockchain.